doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,500 @@
+# Implementing a face beautification algorithm on G-API {#tutorial_gapi_face_beautification}
+
+@prev_tutorial{tutorial_gapi_anisotropic_segmentation}
+
+[TOC]
+
+# Introduction {#gapi_fb_intro}
+
+In this tutorial you are going to learn:
+* How to create a custom kernel implementation (using OpenCV backend);
+* How to use Inference-API to infer networks inside a graph;
+* How to process a videostream inside a graph using Streaming-API;
+* The main ideas of the Face Beautification algorithm.
+
+You can find source code in the `modules/gapi/samples/face_beautification.cpp`
+of the OpenCV source code library.
+
+# New G-API features description {#gapi_fb_features}
+
+Before we can discuss an implementation of any algorithm, we should
+sort out new G-API features applied there. Obviously, paragraphs below are not
+detailed documentary articles, they are just brief guides how to use one feature
+or another with examples of usage.
+
+## Custom kernels implementation {#gapi_fb_custom_kernels}
+
+A new kernel implementation has been already described clearly in OpenCV G-API
+documentation (see [G-API Kernel API](@ref gapi_kernel_api)). Bare steps without
+a lot of details will be mentioned here. In simple words, to implement a brand
+new G-API kernel we have to take 3 main steps:
+
+### 1) New kernel interface definition {#gapi_fb_c_k_1}
+
+There are a couple of macros to define a new kernel. The first is
+`G_TYPED_KERNEL()`, which declares a new operation properties:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp kern_decl
+
+To get the full description, see [Defining a kernel](@ref gapi_defining_kernel).
+
+A little but significant adjustment has to be made when a kernel with a multiple
+return value is going to be defined. For this case, we should use another
+macro, `G_TYPED_KERNEL_M()`:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp kern_m_decl
+
+As we can see, almost everything is unchanged (speaking about the highest
```

Comment:
`As you can see`
`speaking of`
And I would change `highest level` to something more accurate.

Reply:
`highest level` -> `initial look` ?